src/main/java/seedu/address/model/module/UniqueModuleList.java

```diff
@@ -76,6 +77,22 @@ public void setModule(Module target, Module editedModule) {
         internalList.set(index, editedModule);
     }
 
+    /**
+     * Assigns an instructor to the module with the equivalent module code from the list.
+     * The module with the module code must exist in the list.
+     */
+    public void assignInstructor(Person instructor, ModuleCode toAssign) {
+        requireAllNonNull(instructor, toAssign);
+        int indexOfModuleToBeAssigned = 0;
+        while (!internalList.get(indexOfModuleToBeAssigned).hasModuleCode(toAssign)
+                && indexOfModuleToBeAssigned < internalList.size()) {
+            indexOfModuleToBeAssigned++;
+        }
+        Module toSet = internalList.get(indexOfModuleToBeAssigned);
```

Comment:
Do you think it's better if the variable name is changed into "moduleToSet"? 😄

Reply:
I was following the pattern from the other methods (contains, add, remove, etc), let me change the naming for those as well.